packages/compass-data-modeling/src/store/diagram.ts

```diff
@@ -21,12 +25,17 @@ import {
   getDiagramName,
 } from '../services/open-and-download-diagram';
 import type { MongoDBJSONSchema } from 'mongodb-schema';
+import { getCoordinatesForNewNode } from '@mongodb-js/diagramming';
+import { collectionToDiagramNode } from '../utils/nodes-and-edges';
 
 function isNonEmptyArray<T>(arr: T[]): arr is [T, ...T[]] {
   return Array.isArray(arr) && arr.length > 0;
 }
 
-export type SelectedItems = { type: 'collection' | 'relationship'; id: string };
+export type SelectedItems = {
+  type: 'collection' | 'relationship';
+  id?: string;
```

Comment:
It would be really really nice if we could avoid making this optional (just purely looking at how much extra checks we now need to do across all the code). Two suggestions how to maybe handle this better:

- Stop using namespace as an id for collection altogether and assign them an actual unique id similar to what we do for relationships. This clearly separates the collection state from the data modeling metadata
- Assign something reasonably meaningful right away when "creating" a new collection, for example just the database name. This would allow us to avoid at least some of the unnecessary extra "exists" logic, but not ideal because we'd still need to keep those in places where we explicitly check for collection names